cmd/migrate.go

```diff
@@ -116,28 +116,33 @@ func runMigration(_ *cobra.Command, _ []string) error {
 
 	goose.SetBaseFS(assets.EmbedMigrations)
 
-	if version > 0 {
-		currentVersion, err := goose.GetDBVersion(db)
-		if err != nil {
+	currentVersion, err := goose.GetDBVersion(db)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("current version %d", currentVersion)
+
+	if targetVersion == 0 {
+		log.Println("running all migrations")
+		if err := goose.Up(db, migrationsPath); err != nil {
 			log.Fatal(err)
 		}
-
-		int64Version := int64(version)
-		if int64Version < currentVersion {
-			if err := goose.DownTo(db, migrationsPath, int64Version); err != nil {
+	} else {
+		log.Printf("migrating to %d", targetVersion)
+		targetInt64Version := int64(targetVersion)
+		if targetInt64Version < currentVersion {
```

Comment:
Will the most likely case be 
i) no-op
ii) upgrade
iii) downgrade
?

In this case, does it make sense to order the if statement according to this order to optimize path?

Reply:
I don't understand, what would we be optimizing?

Reply:
What I meant is that should we do something like this?

```
   if targetInt64Version == currentVersion {
     // no-op
   } else if targetInt64Version < currentVersion {
     //upgrade
   } else {
     //downgrade
   }
```
as downgrade seldomly happen (save a comparison or two).

Reply:
Migrations are run once in a blue moon.. i don't think that would classify as an optimization